app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/MessageActionsBottomSheetDialog.kt

```diff
@@ -33,45 +33,41 @@ class MessageActionsBottomSheetDialog : ActionsBottomSheetDialog() {
 
     private val navigationArgs: MessageActionsBottomSheetDialogArgs by navArgs()
     private val mainViewModel: MainViewModel by activityViewModels()
-    private val messageActionsViewModel: MessageActionsViewModel by viewModels()
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) = with(binding) {
         super.onViewCreated(view, savedInstanceState)
 
         val threadUid = navigationArgs.threadUid
         val messageUid = navigationArgs.messageUid
 
-        messageActionsViewModel.threadAndMessage(threadUid, messageUid).observe(viewLifecycleOwner) { (thread, message) ->
-
-            if (thread == null || message == null) {
-                findNavController().popBackStack()
-                return@observe
-            }
+        val message = MessageController.getMessage(messageUid) ?: run {
+            findNavController().popBackStack()
+            return
+        }
```

Comment:
Why do it like this compared to before with the viewMode

Reply:
Before we needed to get 2 objects from Realm (the Thread & the Message).

Now we only need to get the Message, so we can directly do it here.

Reply:
keeps the viewModel as before, since we have decided not to make any more exceptions.